muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/announcements-panel.tsx

```diff
@@ -1,69 +1,190 @@
-                      </span>
-                      {extraWorkspaces ? <span className="label">{"(+" + extraWorkspaces + ")"}</span> : null}
-                    </div> : null}
+    const { announcements, currentPage, itemsPerPage } = this.state;
+
+    const offset = currentPage * itemsPerPage;
+
+    /**
+     * Defines current announcements that will be mapped
+     */
+    const currentAnnouncements = announcements.slice(offset, offset + itemsPerPage);
+
+    /**
+     * Calculates amount of pages
+     * depends how many items there is per page
+     */
+    const pageCount = Math.ceil(announcements.length / itemsPerPage);
+
+    /**
+     * renders announcements
+     */
+    const renderAnnouncements = currentAnnouncements.map((announcement: AnnouncementType)=>{
+      const extraWorkspaces = announcement.workspaces && announcement.workspaces.length ? announcement.workspaces.length - 1 : 0;
+      return (
+        <Link key={announcement.id} className={`item-list__item item-list__item--announcements ${announcement.workspaces.length ? "item-list__item--has-workspaces" : ""}`}
+          href={`/announcements#${announcement.id}`} to={`/announcements#${announcement.id}`}>
+          <span className="item-list__icon item-list__icon--announcements icon-paper-plane"></span>
+          <span className="item-list__text-body item-list__text-body--multiline">
+            <span className="item-list__announcement-caption">
+              {announcement.caption}
+            </span>
+            <span className="item-list__announcement-date">
+              {this.props.i18n.time.format(announcement.startDate)}
+            </span>
+            {announcement.workspaces && announcement.workspaces.length ?
+              <div className="labels item-list__announcement-workspaces">
+                <span className="label">
+                  <span className="label__icon label__icon--announcement-workspace icon-books"></span>
+                  <span className="label__text label__text--announcement-workspace">{announcement.workspaces[0].name} {announcement.workspaces[0].nameExtension ? "(" + announcement.workspaces[0].nameExtension + ")" : null }</span>
                 </span>
-              </Link>
-            })}
-          </div>
+                {extraWorkspaces ? <span className="label">{"(+" + extraWorkspaces + ")"}</span> : null}
+              </div> : null}
+          </span>
+        </Link>
+      )
+    })
+
+    /**
+     * renders pagination body as one of announcements list item
+     */
+    const renderPaginationBody = (
+      <div className="item-list__item item-list__item--announcements">
+        <span className="item-list__text-body item-list__text-body--multiline--footer">
+          <ReactPaginate
```

Comment:
There seems to be mixed css classNames/components here. I suggest we use only `pager` components classNames.

Correct structure should be as follows when using our pre-made pager styles:

```
.pager
    .pager__body
        .pager__item pager__item--less icon-arrow-left // If arrow left should be visible
        .pager__item .pager__item--current // If active page
        .pager__item
        .pager__item
        .pager__item pager__item--gap // If there are more pages that gap is needed to display "..."
        .pager__item pager__item--more icon-arrow-right // If arrow right should be visible
```

`.pager__item--first` and `.pager__item--last` might not be applicable with this module. 

ALSO if possible `.pager` could have aria-label added so it follows the same pattern we use in Discussions ie. `aria-label="Tiedotteiden sivutuksen toiminnot/Announcements pagination controls"`